typesafety/test_builtin_severities.yml

```diff
@@ -60,11 +60,11 @@
     logger.error('test {} {} {}', 11, 22, 33, 44)
     logger.exception('{}', 1, 2)
   out: |
-    main:3: note: Expected 0 but found 1 positional arguments for log message
-    main:4: note: Expected 1 but found 3 positional arguments for log message
-    main:5: note: Expected 2 but found 3 positional arguments for log message
-    main:6: note: Expected 3 but found 4 positional arguments for log message
-    main:7: note: Expected 1 but found 2 positional arguments for log message
+    main:3: error: Not all arguments converted during string formatting  [str-format]
```

Comment:
hmm...now looking at it it seems you've managed to take `mypy` to verify formatting on its own, correct? 
If that's the case I think I am taking back what I wrote about changing error messages.
If that's the case we will have `loguru-mypy` aligned to what users of `mypy` experience with string interpolation, which might be desirable.

Reply:
Replying to the following as well:

>  I'd like to hear more about that visit_expr you came up with and how did you that.
>
>  it seems you've managed to take mypy to verify formatting on its own, correct?

Yes that's exactly it. All I do is build a new syntax tree from the existing one, feed it back to mypy, and let it do its thing. Effectively turning

```python
logger.debug('The bar is "{my_foo.bar}"', my_foo=foo)
```

into

```python
logger.debug('The bar is "{my_foo.bar}"'.format(my_foo=foo))
```

to the eyes of mypy.

---

> I really like your job on error messages although I wonder if having them more accurate wasn't better. Could you elaborate why did you change those?
>
>  I think I am taking back what I wrote about changing error messages.
[...] we will have loguru-mypy aligned to what users of mypy experience with string interpolation, which might be desirable.

I agree we are losing a bit of information there, and I'm surprised mypy does not give much to begin with. I was a bit hesitant at first, but if you're cool with it, I'm cool with it :)